drivers/t5838/t5838_nrfx_pdm.c

```diff
@@ -360,29 +405,16 @@ int prv_t5838_aad_unlock_sequence(const struct device *dev)
 
 	if (drv_data->aad_unlocked == false) {
```

Comment:
After powering microphone up, certain sequence needs to be written to enable use of AAD modes. This has to be done every time we want to use AAD after power cycling microphone. 

see section "5.12. AAD ENABLE SEQUENCE" of microphone datasheet.

Reply:
> see section "5.12. AAD ENABLE SEQUENCE" of microphone datasheet.

I missed out this section, so some of my previous comments can be dismissed (most notably my first question).

> It could be that is just safer to reset the mic everytime before user wants to configure it, it that case aad_unlocked would be redundant.

This comment is also wrong, `micen` is optional, so driver should not rely on it.

Copied from the datasheet:
> After this sequence has been completed any of the configuration settings for the AAD Analog or AAD Digital modes can be adjusted. The enable sequence can be written once to the microphone and will remain valid as long as power is maintained to the microphone. If the mic goes through a power cycle, then the sequence will have to be
repeated. 

I suggest you to implement it like that, run the unlock sequence once in the init (if AAD is enabled) or in `t5838_reset` (if AAD is enabled).

Reply:
The issue with running said sequence once during init is that PDM functionality doesn't call init function when zephyr boots like with most zephyr drivers.

API is made in the way that you configure PDM peripheral parameters and call `dmic_configure()` to apply configuration to PDM peripheral.

However, I don't think it is good idea to depend on calling this function as initializer for AAD triggering functionality, since that prevents us from using AAD mode without having configured PDM peripheral. This limits potential use case where we can use AAD for triggering microphone levels but we don't need to actually sample PDM samples.

PDM configuration function is also not good place to do this, since changing PDM sampling rate during runtime does call configuration function multiple times to reapply new configuration. So we would need something to check if we already unlocked AAD modes anyway inside said function.

Reply:
> I suggest you to implement it like that, run the unlock sequence once in the init (if AAD is enabled) or in t5838_reset (if AAD is enabled).

To elaborate my previous statement: I meant `int t5838_pdm_nrfx_init##idx(const struct device *dev)` function, not `dmic_configure`.